src/runtime-rs/crates/hypervisor/src/qemu/qmp.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2024 Red Hat
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::Result;
+use std::fmt::{Debug, Error, Formatter};
+use std::io::BufReader;
+use std::os::unix::net::UnixStream;
+use std::time::Duration;
+
+use qapi::qmp;
+use qapi_spec::Dictionary;
+
+pub struct Qmp {
+    qmp: qapi::Qmp<qapi::Stream<BufReader<UnixStream>, UnixStream>>,
+}
+
+// We have to implement Debug since the Hypervisor trait requires it and Qmp
+// is ultimately stored in one of Hypervisor's implementations (Qemu).
+// We can't do it automatically since the type of Qmp::qmp isn't Debug.
+impl Debug for Qmp {
+    fn fmt(&self, _f: &mut Formatter<'_>) -> Result<(), Error> {
+        Ok(())
+    }
+}
+
+impl Qmp {
+    pub fn new(qmp_sock_path: &str) -> Result<Self> {
+        let stream = UnixStream::connect(qmp_sock_path)?;
+        stream.set_read_timeout(Some(Duration::from_millis(250)))?;
```

Comment:
why is timeout hardcode 250 ?

Reply:
Good question!  The value is admittedly a matter of judgment but there's some thought behind it.  Setting the timeout is to protect runtime-rs from blocking forever trying to set up QMP connection if qemu fails to launch.  

The upper bound is the value of container runtime's task creation timeout (2 seconds for containerd by default) after which the container runtime aborts container launch and terminates the shim anyway.

The lower bound is the maximum delay that can reasonably be expected when things run fine.  I don't know what that could be but since this is a local unix-domain connection I suspect it should be fairly low.  However, it could be noticeable on a system under heavy load, and we don't want to interfere with a slightly delayed but otherwise successful hypervisor launch.

I thought a value between between 100 ms and 250 ms could be good.  I briefly mentioned that to @gkurz and we agreed to put 250 ms.

I'm not insisting that it must be 250 ms, if you have a reason to think that another value would be better I'm open to changing it.

Reply:
Ok, It would be better add a brief comments about it to explain why setting 250 as the value of timeout for developers to check when they're confused.
Anyway, I think the PR reaches maturity. I will approve it.
Thx Pavel @pmores

Reply:
I added a comment with explanation as per your request - please take a look.  Thanks @Apokleos !